### PR TITLE
[Process] [5.0] Replace docblocks by type-hints

### PR DESCRIPTION
**Pull request by tigitz**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | continuation of https://github.com/symfony/symfony/pull/24722 and checks for  #32179
| License       | MIT
| Doc PR        | N/A

This PR adds replace docblocks by type hints in the Process component. Some docblocks without valuable information got also removed.

### CODE REVIEW

File: src/Symfony/Component/Process/ExecutableFinder.php [hunk 1]
```diff
@@ -48,7 +46,7 @@ public function addSuffix($suffix)
      *
      * @return string|null The executable path or default value
      */
-    public function find($name, $default = null, array $extraDirs = [])
+    public function find(string $name, ?string $default = null, array $extraDirs = [])
```
Comment (derrabus) [inline src/Symfony/Component/Process/ExecutableFinder.php]:
```suggestion
    public function find(string $name, string $default = null, array $extraDirs = [])
```

CS: If the parameter has `null` as default value, we omit the `?`.

**Review by derrabus — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Process/Process.php [hunk 2]
```diff
@@ -897,7 +897,7 @@ public function getStatus()
      *
      * @return int The exit-code of the process
      */
-    public function stop($timeout = 10, $signal = null)
+    public function stop($timeout = 10, int $signal = null)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Process.php]: `float` works for the timeout

Comment (tigitz) [reply src/Symfony/Component/Process/Process.php]: Indeed, good catch

**Comment by tigitz** [comment]
@derrabus @nicolas-grekas While my initial scope was to take care of new type-hintable params allowed by 5.0, I went a little bit further and took care of others type-hintable params. But fact is, these additional type-hints can already lands in 4.* IMO and be seen as a continuation of https://github.com/symfony/symfony/pull/24722.

So I'll keep this PR exclusively about it's initial 5.0 scope and make another one targeting 4.* version with these additional type-hints.

Sorry for the bother

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @tigitz.

**Pull request merged**